config/management/src/validator_config.rs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{constants, error::Error, SecureBackends};
+use libra_crypto::{hash::CryptoHash, x25519, ValidCryptoMaterial};
+use libra_network_address::{NetworkAddress, RawNetworkAddress};
+use libra_secure_storage::{Storage, Value};
+use libra_secure_time::{RealTimeService, TimeService};
+use libra_types::{
+    account_address::AccountAddress,
+    transaction::{RawTransaction, SignedTransaction, Transaction},
+};
+use std::{
+    convert::{TryFrom, TryInto},
+    time::Duration,
+};
+use structopt::StructOpt;
+
+// TODO(davidiw) add operator_address, since that will eventually be the identity producing this.
+#[derive(Debug, StructOpt)]
+pub struct ValidatorConfig {
+    #[structopt(long)]
+    owner_address: AccountAddress,
+    #[structopt(long)]
+    validator_address: NetworkAddress,
+    #[structopt(long)]
+    fullnode_address: NetworkAddress,
+    #[structopt(flatten)]
+    backends: SecureBackends,
+}
+
+impl ValidatorConfig {
+    pub fn execute(self) -> Result<Transaction, Error> {
```

Comment:
This is a really long method. Can we not shorten or modularize it? For example, there is some unnecessary duplication here:
- consensus_key, fullnode_network_key and validator_key all have the same logic, except different key names.
- fulnode_address and validator_address have the same logic (different addresses).
- fullnode_network_key and validator_network_key have the same logic (different  keys).